src/crates-io.md

```diff
@@ -0,0 +1,23 @@
+# crates.io Dependencies
+
+The rust compiler supports building with some dependencies from `crates.io`.
+For example, `log` and `env_logger` come from `crates.io`.
+
+In general, you should avoid adding dependencies to the compiler for several
+reasons:
+
+- The dependency may not be high quality or well-maintained, whereas we want
+  the compiler to be high-quality.
+- The dependency may not be using a compatible license.
+- The dependency may have transitive dependencies that have one of the above
+  problems.
+
+TODO: what is the vetting process?
```

Comment:
This is a good question for the meta WG to address, I guess. It seems like a quick FCP *might* make sense.